proxy/src/register.rs

```diff
@@ -0,0 +1,117 @@
+use futures::compat::Future01CompatExt;
+
+use radicle_registry_client::{
+    ed25519::Pair, ClientT, CreateCheckpointParams, DispatchError, Error as ProtocolError,
+    RegisterProjectParams, String32, H256,
+};
+
+/// TODO
+pub struct ProjectParams {
+    /// TODO
+    name: String32,
+    /// TODO
+    domain: String32,
+    /// TODO
+    description: String,
+    /// TODO
+    img_url: String,
+}
+
+/// TODO
+pub enum ClientError {
+    /// TODO
+    Protocol(ProtocolError),
+    /// TODO
+    Runtime(DispatchError),
+}
+
+impl From<ProtocolError> for ClientError {
+    fn from(error: ProtocolError) -> Self {
+        Self::Protocol(error)
+    }
+}
+
+impl From<DispatchError> for ClientError {
+    fn from(error: DispatchError) -> Self {
+        Self::Runtime(error)
+    }
+}
+
+/// TODO
+pub struct Registry<R>
+where
+    R: ClientT,
+{
+    /// TODO
+    client: R,
+}
+
+/// TODO
+impl<R> Registry<R>
+where
+    R: ClientT,
+{
+    /// TODO
+    fn new(client: R) -> Self {
+        Self { client }
+    }
+
+    /// TODO
+    async fn register_project(
```

Comment:
We should not jump into `async` just yet. It seems juniper will not play nicely. So for now we should use the blocking calls on the client with `wait` and later on `block_on` once radicle-dev/radicle-registry#138 is done.

Reply:
Yes, I found this out the hard way. Sounds good.